### ISSUE TITLE
How to copy the whole DB data/schema from one machine to another?

### ISSUE THREAD

**Issue by Jack012a** [ClickHouse/ClickHouse#23735]
Hello all,

I have two machines(i.e. A and B) and machine A has ClickHouseDB installed.
Now I need to duplicate the same setup(data/schema) from machine A to machine B.

Can I copy all directories under /var/lib/clickhouse/ from machine A to machine B?

```
$ ls -l /var/lib/clickhouse/
total 28
drwxr-xr-x 2 clickhouse clickhouse 4096 Apr 27 10:35 cores
drwxr-xr-x 4 clickhouse clickhouse 4096 Apr 27 10:35 data
drwxr-xr-x 2 clickhouse clickhouse 4096 Apr 27 10:35 flags
drwxr-xr-x 2 clickhouse clickhouse 4096 Apr 27 10:35 format_schemas
drwxr-xr-x 4 clickhouse clickhouse 4096 Apr 27 10:35 metadata
drwxr-xr-x 2 clickhouse clickhouse 4096 Apr 27 10:35 tmp
drwxr-xr-x 2 clickhouse clickhouse 4096 Apr 27 10:35 user_files
```

Thank you

**Comment by nikitamikhaylov** [comment]
1) If you have a little data on server A, you can simply copy the entire `/var/lib/clickhouse` folder to server B. But be careful with access rights (I mean user and group), they are must be `clickhouse`. Also this approach won't work if you have a little bit tricky multi disks configuration. 
2) You can create the same database and tables on server B and then copy each table with `INSERT SELECT` query and `remote` table function
3) If you have large amount of data and quite big partitions, you can use `clickhouse-copier`. See references for details.
https://clickhouse.tech/docs/en/operations/utilities/clickhouse-copier/ 
https://altinity.com/blog/2018/8/22/clickhouse-copier-in-practice

**Comment by praisethejack** [comment]
@nikitamikhaylov I know the author is well-meaning and should really want to make it clear, the fact is that the two tutorials on point 3 are hard to follow

**Comment by alexey-milovidov** [comment]
Use BACKUP and RESTORE commands for that.

**Comment by lampjian** [comment]
clickhouse-copier works well until you share the zookeeper configuration with the source cluster. There is no difference between independent  and shared zookeeper clusters.

**Comment by lampjian** [comment]
BACKUP and RESTORE commands is available from version 21.10 and later.
